### PR TITLE
Consolidate num_traits foundation parity into a generic carrier harness

### PR DESCRIPTION
**Pull request by kaidokert**
Test-only. Closes the consolidation debt noted after PR #148: its num_traits foundation parity (Zero/One/Bounded/ToPrimitive/FromPrimitive/NumCast) was written as bespoke inline tests against one carrier, not the `for_both_carriers` template we use elsewhere.

**What:**
- New `tests/carrier_num_traits.rs` — the `feature = "num-traits"` counterpart to `carrier_generic.rs` (which stays feature-free for the const-only surface, hence a separate file rather than adding num_traits bounds to the ungated one). A `NumCarrier` trait bundles the foundation, one generic body per concern, run for both carriers across three 32-bit backings.
- `num_traits_bridge.rs` inline tests trimmed to only what the fixed-width harness structurally can't reach: `from_u64`'s natural-width construction (heapless-specific) and a wide (>32-bit) multi-limb round-trip that also exercises the >u64-max `Bounded` path.

No production code changes. Harness gates out cleanly with `--no-default-features`; clippy clean.

## Summary by Sourcery

Introduce a shared num_traits test harness for both fixed-width and heapless carriers while trimming heapless-specific inline tests to only cover behavior not exercised by the generic harness.

New Features:
- Add a feature-gated carrier_num_traits test module that validates num_traits foundation (Zero/One/Bounded/ToPrimitive/FromPrimitive/NumCast) across both carriers and multiple 32-bit backings.

Enhancements:
- Refocus heapless num_traits bridge tests on natural-width construction and wide multi-limb round-trip cases that fall outside the fixed-width carrier harness.

Tests:
- Add generic NumCarrier trait and for_both_carriers harness to deduplicate and standardize num_traits coverage for FixedUInt and HeaplessBigInt.
- Remove now-redundant num_traits foundation tests from num_traits_bridge.rs, keeping only heapless-specific behavior checks.

## Summary by CodeRabbit

* **Tests**
  * Expanded numeric trait coverage across supported fixed-width and heapless integer types.
  * Added shared generic assertions covering zero/one and bounded min/max behavior across multiple 32-bit backings.
  * Verified primitive conversion and casting for in-range `u64` values, with rejection for values outside the carrier’s width.
  * Improved wide-value round-trip coverage, including correct handling when the max value cannot fit in `u64`.

### CODE REVIEW

File: src/heapless/num_traits_bridge.rs [hunk 1]
```diff
@@ -187,4 +136,14 @@ mod tests {
...
     }
+
+    #[test]
+    fn wide_value_roundtrip() {
+        // Values wider than 32 bits (multi-limb u64 assembly, and the
+        // >u64-max `Bounded` path where to_u64 of the 256-bit max is None)
+        // round-trip in a carrier wide enough to hold them.
+        for v in [0x1_0000_0000u64, 0x1234_5678_9ABC, 0xFFFF_FFFF_FFFF_FFFF] {
+            assert_eq!(H32x8::from_u64(v).unwrap().to_u64(), Some(v));
+        }
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/heapless/num_traits_bridge.rs:146]: @gemini-code-assist Added in 0745b34 — the test now asserts `<H32x8 as num_traits::Bounded>::max_value().to_u64() == None` explicitly, so the branch from_u64 relies on is pinned rather than only implied.
